File: pingyougou/pingyougou-manage-web/src/main/java/manage/controller/BrandController.java
package manage.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import entity.TbBrand;
import org.springframework.web.bind.annotation.*;
import sellergoods.service.BrandService;
import vo.PageResult;
import vo.Result;

import java.util.List;
import java.util.Map;

@RequestMapping("/brand")
@RestController
public class BrandController {

    //@Reference表示该用户（变量）就是消费者（在springMVC中有详细配置）
    @Reference
    private BrandService bs;

    @GetMapping("/findAll")
    public List<TbBrand> queryAll(){
        System.out.println("-------------------------------------------------");
        /*List<TbBrand> list = bs.findAllBrand();
        System.out.println(list);
        return list;*/
        return bs.findAll();
    }

    @GetMapping("/testPage")
    public List<TbBrand> testPage(@RequestParam(defaultValue = "1") Integer pageNum,
                                  @RequestParam(defaultValue = "10") Integer pageSize){
        //常规路径
        //return bs.testPage(pageNum,pageSize);

        //通用mapper路径
        return (List<TbBrand>) bs.findPage(pageNum,pageSize);
    }

    @GetMapping("/findPage")
    public PageResult findPage(@RequestParam(value = "page",defaultValue = "1")Integer pageNum,
                                @RequestParam(value = "rows",defaultValue = "10")Integer pageSize){
        return bs.findPage(pageNum,pageSize);
    }

    //添加
    @PostMapping("/add")
    public Result add(@RequestBody TbBrand brand){
        try{
            bs.add(brand);
            return Result.ok("添加成功");
        }catch (Exception e){
            e.printStackTrace();
        }
        return Result.fail("添加失败");
    }

    //修改
    @GetMapping("/findOne")
    public TbBrand findOne(Long id){
        return bs.findOne(id);
    }

    @PostMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try{
            bs.update(brand);
            return Result.ok("添加成功");
        }catch (Exception e){
            e.printStackTrace();
        }
        return Result.fail("添加失败");
    }

    //批量删除
    @GetMapping("/delete")
    public Result delete(Long[] ids){
        System.out.println("delete  ================================================================ "+ids);
        try{
            bs.deleteByIds(ids);
            return Result.ok("删除成功");
        }catch (Exception e){
            e.printStackTrace();
        }
        return Result.fail("删除成功");
    }

    //条件查询
    @PostMapping("/search")
    public PageResult search(@RequestBody TbBrand brand,
                             @RequestParam(value = "page",defaultValue = "1")Integer pageNum,
                             @RequestParam(value = "rows",defaultValue = "10")Integer pageSize){
        return bs.search(brand,pageNum,pageSize);
    }

    //查询品牌列表，返回的数据格式符合select2格式
    @GetMapping("/selectOptionList")
    public List<Map<String,Object>> selectOptionList(){
        return bs.selectOptionList();
    }

}

File: pingyougou/pingyougou-manage-web/src/main/webapp/js/controller/brandController.js
//表现层


app.controller("brandController", function ($scope, $http ,$controller,brandService) {
    

    //继承baseController
    $controller("baseController",{$scope:$scope});

    //搜索条件
    $scope.searchEntity = {};

    //分页查询
    $scope.findPage = function (page, rows) {
        brandService.findPage(page, rows).success(function (response) {
            alert("response.rows = "+response.rows);
            //更新记录列表
            $scope.list = response.rows;

            alert("response.total = "+response.total);

            //更新总记录数
            $scope.paginationConf.totalItems = response.total;

        }).error(function () {
            alert("数据加载失败!");
        })
    };

    //查询所有列表数据并绑定到list对象,查询品牌列表
    $scope.findAll = function () {
        //get方式异步提交到后台（ajax）
        brandService.findAll().success(function (response) {
            $scope.list = response;
        });
    };

    //保存
    $scope.save = function () {
        var obj;
        if ($scope.entity.id != null) {
            obj = brandService.update($scope.entity);
        } else {
            obj = brandService.add($scope.entity);
        }
        obj.success(function (response) {
            if(response.success){
                //重新加载列表
                $scope.reloadList();
            }else {
                alert(response.message);
            }
        });
    };

    //根据主键查询
    $scope.findOne = function (id) {
        brandService.findOne(id).success(function (response) {
            $scope.entity = response;
        });
    };

    //批量删除
    $scope.delete = function () {
        if($scope.selectedIds.length < 1){
            alert("请选择要删除的记录");
            return ;
        }
        if(confirm("确定要删除所选记录吗？")){
            brandService.delete($scope.selectedIds).success(function (response) {
                if(response.success){
                    $scope.reloadList();
                    $scope.selectedIds = [];
                }else{
                    alert(response.message);
                }
            });
        }
    };

    //搜索框
    $scope.search = function (page,rows) {
        brandService.search(page,rows,$scope.searchEntity).success(function (response) {
            $scope.list = response.rows;
            $scope.paginationConf.totalItems = response.total;
        });
    };
});